corehq/apps/users/tests/test_clone_web_users.py

```diff
@@ -0,0 +1,127 @@
+            None,
+            USER_CHANGE_VIA_CLONE,
+            by_domain_required_for_log=False,
+            email='new-user@dimagi.com',
+        )
+
+    def tearDown(self):
+        self.old_user.delete(self.domain, deleted_by=None)
+        self.new_user.delete(self.domain, deleted_by=None)
+        super(TestCloneWebUsers, self).tearDown()
+
+    def test_copy_domain_memberships(self):
+        self.assertEqual(1, len(self.old_user.domain_memberships))
+        self.assertEqual([], self.new_user.domain_memberships)
+
+        copy_domain_memberships(self.old_user, self.new_user)
+
+        self.assertEqual(len(self.old_user.domain_memberships), len(self.new_user.domain_memberships))
+
+    def test_copy_domain_memberships_copies_roles(self):
+        old_domain_membership = self.old_user.domain_memberships[0]
+        old_domain_membership.role_id = 'abc123'
+
+        copy_domain_memberships(self.old_user, self.new_user)
+
+        new_domain_membership = self.new_user.domain_memberships[0]
+        self.assertEqual('abc123', new_domain_membership.role_id)
+
+    def test_copy_domain_memberships_copies_locations(self):
+        old_domain_membership = self.old_user.domain_memberships[0]
+        old_domain_membership.location_id = 'abc123'
+
+        copy_domain_memberships(self.old_user, self.new_user)
+
+        new_domain_membership = self.new_user.domain_memberships[0]
+        self.assertEqual('abc123', new_domain_membership.location_id)
+
+    def test_transfer_exports(self):
+        export = FormExportInstance(owner_id=self.old_user._id, domain=self.domain)
+        export.save()
+        self.addCleanup(export.delete)
+
+        transfer_exports(self.old_user, self.new_user)
+
+        export = _get_export_instance(ExportInstance, [self.domain])[0]
+        self.assertEqual(export.owner_id, self.new_user._id)
+
+    def test_transfer_scheduled_reports(self):
+        scheduled_report = ReportNotification(owner_id=self.old_user._id, domain=self.domain)
+        scheduled_report.save()
+        expected_id = scheduled_report._id
+        self.addCleanup(scheduled_report.delete)
+
+        transfer_scheduled_reports(self.old_user, self.new_user._id)
+
+        actual_id = ReportNotification.by_domain_and_owner(self.domain, self.new_user._id, stale=False)[0]._id
+        self.assertEqual(expected_id, actual_id)
+
+    def test_transfer_saved_reports(self):
+        saved_report = ReportConfig(
+            name='test',
+            owner_id=self.old_user._id,
+            report_slug='worker_activity',
+            report_type='project_report',
+            domain=self.domain)
+        saved_report.save()
+        expected_id = saved_report._id
+        self.addCleanup(saved_report.delete)
+        transfer_saved_reports(self.old_user, self.new_user)
+
+        actual_id = ReportConfig.by_domain_and_owner(self.domain, self.new_user._id, stale=False)[0]._id
+        self.assertEqual(expected_id, actual_id)
```

Comment:
Thanks for mentioning the flaky test possibility. It should have crossed my mind when thinking about that solution, and agree it would make the test less readable. I set stale to False again in the saved report and scheduled report queries. I think about adding it as an optional param defaulted to True that tests could override, but given that this is a management command that isn't run often, I think it is best to have stale set to false anyway.